packages/vizbuilder/src/helpers/chartconfig.js

```diff
@@ -0,0 +1,330 @@
+      y: measureName,
+      yConfig: {title: measureName},
+      stacked: drilldown.depth > 1
+    };
+
+    if (timeDrilldown) {
+      config.groupBy = [timeDrilldown.hierarchy.levels[1].name];
+    }
+
+    if (flags.availableKeys.has("State")) {
+      config.xSort = sortByGeographyState;
+    }
+    else {
+      config.xSort = sortByCustomKey(drilldownName);
+    }
+
+    return config;
+  },
+  barchartyear(commonConfig, query) {
+    const {drilldown, timeDrilldown, measure} = query;
+
+    const drilldownName = timeDrilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      discrete: "x",
+      x: drilldownName,
+      xConfig: {title: drilldownName},
+      y: measureName,
+      yConfig: {title: measureName},
+      stacked: true,
+      groupBy: [drilldown.name],
+      tooltipConfig: tooltipConfig(query, drilldownName)
+    };
+
+    return config;
+  },
+  histogram(commonConfig, query, flags) {
+    // HISTOGRAM
+    // Allows to see the frequency distribution of a continuous dataset.
+    // Useful mainly with range buckets.
+
+    const config = this.barchart(commonConfig, query, flags);
+    config.groupPadding = 0;
+    return config;
+  },
+  donut(commonConfig, query) {
+    const {drilldown, measure} = query;
+
+    const drilldownName = drilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      xConfig: {title: null},
+      y: measureName,
+      yConfig: {title: measureName},
+      groupBy: drilldownName
+    };
+
+    return config;
+  },
+  geomap(commonConfig, query, flags) {
+    const drilldownName = query.drilldown.name;
+
+    const config = {
+      ...commonConfig,
+      ...flags.colorScale,
+      tiles: false,
+      id: `ID ${drilldownName}`,
+      groupBy: `ID ${drilldownName}`,
+      ocean: "transparent",
+      projection: "geoAlbersUsa",
+      topojson: "/topojson/states.json",
+      topojsonId: "id",
+      topojsonKey: "states",
+      zoom: true,
+      zoomFactor: 2,
+      zoomScroll: false
+    };
+
+    if (flags.activeType === "geomap") {
+      const size = getAreaChartDimensions();
+      config.colorScaleConfig = {
+        scale: "jenks",
+        height: size.height,
+        width: size.width
+      };
+    }
+
+    return config;
+  },
+  lineplot(commonConfig, query) {
+    const {timeDrilldown, drilldown, measure, moe} = query;
+
+    const drilldownName = timeDrilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      discrete: "x",
+      groupBy: drilldown.hierarchy.levels[1].name,
+      x: drilldownName,
+      xConfig: {title: drilldownName},
+      y: measureName,
+      yConfig: {title: measureName}
+    };
+
+    if (moe) {
+      const moeName = moe.name;
+
+      config.confidence = [
+        d => d[measureName] - d[moeName],
+        d => d[measureName] + d[moeName]
+      ];
+      config.confidenceConfig = {
```

Comment:
this `confidenceConfig` should be user-definable, please remove.

Reply:
removed in 096b8f5a0900b84d4341cca21e35db479343399a